db/version_set.cc

```diff
@@ -1120,7 +1120,7 @@ void Version::UpdateAccumulatedStats(bool update_stats) {
           // when option "max_open_files" is -1, all the file metadata has
           // already been read, so MaybeInitializeFileMetaData() won't incur
           // any I/O cost.
-          if (vset_->db_options_->max_open_files == -1) {
+          if (vset_->mutable_db_options_->max_open_files == -1) {
```

Comment:
Not familiar with the logic, but do we hold db_mutex here? mutable_db_options is guarded by db_mutex.